lib/src/components/range_selector/range_selector.dart

```diff
@@ -0,0 +1,455 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+import '../../../zeta_flutter.dart';
+
+/// The [ZetaRangeSelector] is a customizable range selector widget that
+/// allows users to select a range of values within a specified minimum
+/// and maximum range. It provides a visual representation of the selected
+/// range and allows for precise adjustments through both dragging and direct input
+class ZetaRangeSelector extends ZetaStatefulWidget {
+  /// Creates a new [ZetaRangeSelector]
+  ZetaRangeSelector({
+    super.key,
+    super.rounded,
+    required this.initialValues,
+    this.min = 0.0,
+    this.max = 100,
+    this.label,
+    this.onChange,
+    this.divisions,
+    this.semanticLabel,
+    this.showValues = true,
+    this.largeSliderThumbs = false,
+  }) : assert(
+          min <= initialValues.start && initialValues.start <= initialValues.end && initialValues.end <= max,
+          'The start value must be less than or equal to the end value, and both must be within the range of min and max.',
+        );
+
+  /// The initial values of the range selector.
+  final RangeValues initialValues;
+
+  /// The minimum value of the range selector.
+  /// Defaults to 0.
+  final double min;
+
+  /// The maximum value of the range selector.
+  /// Defaults to 100.
+  final double max;
+
+  /// The label of the range selector.
+  final String? label;
+
+  /// Called with the values of the range selector whenever it is changed.
+  final ValueChanged<RangeValues>? onChange;
+
+  /// The number of divisions for the range selector.
+  final int? divisions;
+
+  /// Value passed to the wrapping [Semantics] widget.
+  final String? semanticLabel;
+
+  /// Whether to show the values of the range selector.
+  /// Defaults to true.
```

Comment:
In general we don't _need_ to have 'Defaults to true' in the comments - the flutter analyzer knows the defaults and can pre populate this in any documentation 
<img width="472" alt="image" src="https://github.com/user-attachments/assets/f67c8b66-3c16-4264-822e-ff30c381d971">